android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/Utils.java

```diff
@@ -58,6 +62,24 @@
   private static final String PWSTRIPTIME_KEY = "pwstriptime";
   private static final RegionResolver REGION_RESOLVER = new RegionResolver();
   private static final String SEPARATOR = "\0";
+  public static Comparator<PwPair> mDistanceSort = new Comparator<PwPair>(){
```

Comment:
put this behind a method that creates a fresh one each time (so that we have a new set of cached distances)